app.py

```diff
@@ -16,94 +16,191 @@
 from pytimeparse.timeparse import timeparse
-import requests
+
+from prometheus_client import CollectorRegistry, Gauge, Counter, push_to_gateway
 
 from thoth.common import init_logging
 from thoth.common import get_service_account_token
+from thoth.common import __version__ as __common__version__
+
+
+__version__ = f"0.6.1+common.{__common__version__}"
+
 
+init_logging()
+prometheus_registry = CollectorRegistry()
 
 _LOGGER = logging.getLogger('thoth.cleanup_job')
 
 KUBERNETES_API_URL = os.getenv('KUBERNETES_API_URL', 'https://kubernetes.default.svc.cluster.local')
 KUBERNETES_API_TOKEN = os.getenv('KUBERNETES_API_TOKEN') or get_service_account_token()
+KUBERNETES_VERIFY_TLS = bool(int(os.getenv('KUBERNETES_VERIFY_TLS', 0)))
 THOTH_MIDDLETIER_NAMESPACE = os.environ['THOTH_MIDDLETIER_NAMESPACE']
-THOTH_ANALYZER_CLEANUP_TIME = timeparse(os.getenv('THOTH_ANALYZER_CLEANUP_TIME', '7d'))
+THOTH_CLEANUP_TIMEOUT = timeparse(os.getenv('THOTH_CLEANUP_TIMEOUT', '5d'))
+THOTH_METRICS_PUSHGATEWAY_URL = os.getenv('THOTH_METRICS_PUSHGATEWAY_URL')
+THOTH_MY_NAMESPACE = os.getenv("NAMESPACE", "thoth-test-core")
 
+_METRIC_RUNTIME = Gauge(
+    'thoth_cleanup_job_runtime_seconds', 'Runtime of cleanup job in seconds.', [],
+    registry=prometheus_registry)
+_METRIC_SOLVER_JOBS = Counter(
+    'thoth_cleanup_job_solver_jobs', 'Solver Jobs cleaned up.', ['env', 'op'],
+    registry=prometheus_registry)
 
-def _get_analyzers():
-    """Get currently running analyzers."""
+# Metrics Exporter Metrics
+_METRIC_INFO = Gauge('thoth_cleanup_job_info',
+                     'Thoth Cleanup Job information', ['env', 'version'], registry=prometheus_registry)
+_METRIC_INFO.labels(THOTH_MY_NAMESPACE, __version__).inc()
+
+
+def _get_pods(label_selector: str) -> dict:
+    """Get currently running analyzers by label."""
     # TODO: pagination?
-    endpoint = "{}/{}/{}/{}".format(KUBERNETES_API_URL,
-                                    'api/v1/namespaces',
-                                    THOTH_MIDDLETIER_NAMESPACE,
-                                    'pods?labelSelector=thothtype%3Duserpod')
+    endpoint = (f"{KUBERNETES_API_URL}/api/v1/namespaces/{THOTH_MIDDLETIER_NAMESPACE}/"
+                f"pods?labelSelector={label_selector}")
+
     response = requests.get(
         endpoint,
-        verify=bool(int(os.getenv('KUBERNETES_VERIFY_TLS', True))),
+        verify=KUBERNETES_VERIFY_TLS,
         headers={
             'Authorization': 'Bearer {}'.format(KUBERNETES_API_TOKEN),
             'Content-Type': 'application/json'
         }
     )
     response.raise_for_status()
     _LOGGER.debug('Full response from Kubernetes master is: %r', response.json())
+
     pods = response.json().get('items', [])
-    _LOGGER.info('Kubernetes master returned %d pods', len(pods))
+    _LOGGER.info('Kubernetes master returned %r pods with label %r', len(pods), label_selector)
+
     return pods
 
 
+def _get_jobs(label_selector: str) -> dict:
```

Comment:
We have this logic in common - https://github.com/thoth-station/common/blob/b199f75c93625ffc836482d002f8921f1a029528/thoth/common/openshift.py#L319 Also it would be great to have the deletion at one place as well.